tests/grid/grid_tools_aspect_ratio.cc

```diff
@@ -0,0 +1,205 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2001 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Short test to validate GridTools::compute_maximum_aspect_ratio()
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/partitioner.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/mapping_q_generic.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+template <int dim>
+double
+compute_aspect_ratio_hyper_rectangle(
+  Point<dim> const &               left,
+  Point<dim> const &               right,
+  std::vector<unsigned int> const &refinements,
+  unsigned int                     degree     = 1,
+  unsigned int                     n_q_points = 2,
+  bool                             deform     = false,
+  double                           factor     = 1.0)
```

Comment:
Can you add a short comment for this function that discusses the purpose of the parameters, in particular `refinements`, `deform` and `factor`, so in case some else stumbles across this test immediately sees the purpose? While there, you might also want to output the case that is currently done (i.e., the value of these three parameters plus maybe the dimension that you can then remove from the main function) at the top of this function. Imagine a case where one test fails - one wants to immediately spot in the output file what configuration has failed, rather than counting the calls in the main function and compare to the output file.

Also, I think you can skip the `degree` parameter` because you only have a flat manifold where `MappingQGeneric` degenerates to a bi-/trilinear mapping in any case. The number of quadrature points is more interesting, and I would suggest that add two more calls to this function where you pick more than 2 points, say for the 2D cases with moderate deformed vertex (where I would expect a different result for the aspect ratio) and one without deformation (where I would expect the same result as previously.

Reply:
done in https://github.com/dealii/dealii/pull/8894/commits/309e90b0631f3c3a4a40bf7380addaab683dd01a